spec/models/waste_exemptions_engine/address_spec.rb

```diff
@@ -18,45 +18,50 @@ module WasteExemptionsEngine
 
     context "scopes" do
       before do
-        # TODO: This is necessary as those tests are generating random failures due to
+        # TODO: This is necessary as these tests are generating random failures due to
         # the database not being cleaned properly by some other test
         described_class.delete_all
       end
 
-      describe ".missing_easting_or_northing" do
-        it "returns all address with x and y information" do
+      describe ".sites_missing_easting_or_northing" do
+        it "returns all site addresses missing x and y information" do
           missing_info_records = []
-          missing_info_records << create(:address, x: nil, y: 123.4)
-          missing_info_records << create(:address, x: 123.4, y: nil)
-          missing_info_records << create(:address, x: nil, y: nil)
+          missing_info_records << create(:address, :site_address, x: nil, y: 123.4)
+          missing_info_records << create(:address, :site_address, x: 123.4, y: nil)
+          missing_info_records << create(:address, :site_address, x: nil, y: nil)
 
           create(:address, x: 123.4, y: 123.4)
+          create(:address, x: nil, y: nil)
+          create(:address, :site_address, x: 123.4, y: 123.4)
 
-          expect(described_class.missing_easting_or_northing).to match_array(missing_info_records)
+          expect(described_class.sites_missing_easting_or_northing).to match_array(missing_info_records)
         end
       end
-      describe ".with_easting_and_northing" do
-        it "returns all address with x and y information" do
+      describe ".sites_with_easting_and_northing" do
+        it "returns all site addresses with x and y information" do
           create(:address, x: nil, y: 123.4)
-          create(:address, x: 123.4, y: nil)
+          create(:address, :site_address, x: 123.4, y: nil)
```

Comment:
Not a showstopper, but for completeness could we also have:
- a non-site address missing a northing
- a site address missing an easting?

Reply:
![stormoff](https://user-images.githubusercontent.com/1789650/65697847-8fcdc980-e073-11e9-9e1b-1260b10e248d.gif)

Reply:
You've gone above and beyond 👍